plugins/modules/aci_aaa_domain.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Sabari Jaganathan (@sajagana) <sajagana@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "certified"}
+
+DOCUMENTATION = r"""
+---
+module: aci_aaa_domain
+short_description: Manage AAA domains (aaa:Domain)
+description:
+- Manage AAA domains on Cisco ACI fabrics.
+options:
+  name:
+    description:
+    - The name of the aaa domain.
+    type: str
+    aliases: [ aaa_domain ]
+  description:
+    description:
+    - Description of the aaa domain.
+    type: str
+    aliases: [ descr ]
+  restricted_rbac_domain:
+    description:
+    - C(True) to enable Restricted RBAC Domain on the aaa security domain.
+    type: bool
+    choices: [ false, true ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+  name_alias:
```

Comment:
what dose name_alias specific mean?

Reply:
name_alias mapped with nameAlias field in the backend API side.